### ISSUE TITLE
MySQL: orm should support context

### ISSUE THREAD

**Issue by scbizu** [ezbuy/redis-orm#84]
## background

In our daily development, we use the SQL operations together with some rpc functions . And, we pass the context between the rpc invocation , however, redis-orm do not accept this ctx within its SQL execution.

## proposal 

```go
	result, err := tx.tx.Exec(sql, args...)
	if err != nil {
		tx.err = err
	}
```

we should provide a new method to pass the outside ctx into the `Exec` , perhaps it will be like : 

```go 
	result, err := tx.tx.ExecContext(ctx,sql, args...)
	if err != nil {
		tx.err = err
	}
```

## covered situation

* grpc: The invoker inspector client canceled(OR Timeout) the grpc request,and set the context,but the server side will continue the SQL operations .And once the invoker retries , it will receive some database error(i.e.  the record lock error OR the dirty reads)

**Comment by 54niyu** [comment]
It is a good idea !

**Comment by scbizu** [comment]
see #86

**Comment by scbizu** [comment]
Fixed in #86
